mycompany-seeq-link-connector-myconnector/src/main/java/com/mycompany/seeq/link/connector/MyConnection.java

```diff
@@ -225,10 +257,89 @@ public Integer getMaxResultsPerRequest() {
         return this.connectionConfig.getMaxResultsPerRequest();
     }
 
+    @Override
+    public Stream<Capsule> getCapsules(GetCapsulesParameters parameters) throws Exception {
+        if (parameters.isLastCertainKeyRequested()) {
+            TimeInstant endTime = new TimeInstant(ZonedDateTime.now());
+            TimeInstant startTime = new TimeInstant(endTime.getTimestamp() - parameters.getMaxDuration());
+            DatasourceSimulator.TagValue lastTagValue = this.datasourceSimulator.requestLastTagValue(
+                    parameters.getDataId(),
+                    startTime,
+                    endTime
+            );
+
+            if (lastTagValue != null) {
+                parameters.setLastCertainKey(new TimeInstant(lastTagValue.getEnd()));
```

Comment:
You don't need to do this unless there is some reason that the cursor needs to be different than the lastTagValue, the cursor will automatically be set to the last tag value in the SDK.